model/relabel/relabel.go

```diff
@@ -205,6 +205,11 @@ func (re Regexp) MarshalYAML() (interface{}, error) {
 	return nil, nil
 }
 
+// IsZero implements the yaml.IsZeroer interface.
+func (re Regexp) IsZero() bool {
+	return re.Regexp == DefaultRelabelConfig.Regex.Regexp
```

Comment:
I assume this takes into account cases where the default `(.*)` regex is explicitly provided, as we compare pointers here.
It'd be great to have a regression test for it though.

Reply:
The test shows that this change is not good... Because the input and the marshalled should be different then

Reply:
I'm not sure I follow, if the regex is not provided in the original yaml, then when we unmarshal and marshal again it will still not be in the resulting yaml. If the default regex is explicitly provided then when we unmarshal and marshal it will still be there explicitly in the yaml, is that not expected?